src/model/misc.rs

```diff
@@ -53,6 +51,108 @@ impl_from_str! {
     ChannelId, ChannelIdParseError, parse_channel;
 }
 
+/// An image hash returned from the Discord API.
+///
+/// Note: This is parsed into a compact form when constructed, then turned back
+/// into the cannonical hex representation used by the API when needed.
```

Comment:
Speaking of my previous comment, it is not immediately clear how this type should be constructed by the user if it is going to be passed down as an argument to functions like `Guild::create`. We should mention the `FromStr` implementation here with an example.